client/pages/edit-joke-form.jsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+
+export default class EditJokeForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      joke: '',
+      title: '',
+      approxMinutes: '',
+      categoryId: '',
+      categories: []
+    };
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleChangeTitle = this.handleChangeTitle.bind(this);
+    this.handleChangeJoke = this.handleChangeJoke.bind(this);
+    this.handleChangeApproxMinutes = this.handleChangeApproxMinutes.bind(this);
+    this.categoryList = this.categoryList.bind(this);
+    this.categorySelect = this.categorySelect.bind(this);
+  }
+
+  componentDidMount() {
+    fetch('/api/jokeApp/categories')
+      .then(res => res.json())
+      .then(categories => this.setState({ categories }));
+  }
+
+  handleChangeTitle(event) {
+    this.setState({ title: event.target.value });
+  }
+
+  handleChangeJoke(event) {
+    this.setState({ joke: event.target.value });
+  }
+
+  handleChangeApproxMinutes(event) {
+    this.setState({ approxMinutes: event.target.value });
+  }
+
+  categorySelect(event) {
+    this.setState({ categoryId: event.target.value });
+  }
```

Comment:
You should only need one change handler for your inputs. If you put a `name` attribute on each input thats value matches the property name in state you can use that to set state. For example:

```JSX
handleChange(event) {
  const name = event.target.name;
  const value = event.target.value;
  
  this.setState({
    [name]: value
  });
}
```